src/utils/management/CustomState.ts

```diff
@@ -1,34 +1,23 @@
-export class CustomState<T> {
-  private error: string;
-  private pending: boolean;
-  private data: T;
-  private readonly preparedData: T;
+export function State<T>(data: T): CustomState<T> {
+  return new CustomState(data);
+}
 
-  constructor(preparedData: T) {
-    this.preparedData = preparedData;
-  }
+class CustomState<T> {
+  error: string;
+  pending: boolean;
+  preparedData: T;
+
+  constructor(public data: T) {}
 
-  public init(): CustomState<T> {
-    const copy: CustomState<T> = new CustomState<T>(this.preparedData);
-    copy.pending = true;
-    copy.data = this.preparedData;
-    copy.error = '';
-    return copy;
+  init(): CustomState<T> {
```

Comment:
Refactor init, ok, fail to be arrow functions. 

Example:

const ok = <T>(data: T) => ({ ...this, data, error: '', pending: false });

There is one main reason why. With normal functions without bind usage in the constructor, we can have a messy bug here. Imagine situation:

const state = State<YourType>(INITIAL_SHAPE);
state.ok.call(pass here other this for example window) - program will break.

The same problem occurs with implicit this binding.